### PR TITLE
[rb] Use id in manifest json for firefox extension if exists

### PR DESCRIPTION
**Pull request by tadashi0713**
- [ X ] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

Try to solve this issue
https://github.com/SeleniumHQ/selenium/issues/6471

Similar modification as python
https://github.com/SeleniumHQ/selenium/pull/6463

---
This change is [Reviewable](https://reviewable.io/reviews/seleniumhq/selenium/6550)

### CODE REVIEW

File: rb/lib/selenium/webdriver/firefox/extension.rb [hunk 1]
```diff
@@ -87,7 +87,10 @@ def read_id_from_manifest_json(directory)
           return unless File.exist?(manifest_path)
 
           manifest = JSON.parse(File.read(manifest_path))
-          [manifest['name'].delete(' '), manifest['version']].join('@')
+          id = if manifest.key?('application') && manifest['application'].key?('gecko')
```
Comment (luke-hill) [inline rb/lib/selenium/webdriver/firefox/extension.rb]: Couldn't you just write this as an if/else or a guard clause like on line 87? Just seems a bit weird doing this conditional assignment for essentially an if/else.

Comment (tadashi0713) [reply rb/lib/selenium/webdriver/firefox/extension.rb]: @luke-hill Thanks for advice

**Comment by davehunt** [comment]
@titusfortner I agree this should be tested as it's the more common use case. It should be trivial to add a web extension with an ID for testing purposes.

**Comment by p0deje** [comment]
I couldn't merge this PR because it needed code changes (e.g. look for `applications`, not `application`), so I've pushed a25638dfb68e2752a224ab7716cb5edc491e4033 to fix this.

In any event, thank you for the proposed contribution!

**Pull request closed without merging**